staging/src/k8s.io/client-go/tools/watch/until.go

```diff
@@ -86,7 +86,7 @@ func UntilWithoutRetry(ctx context.Context, watcher watch.Interface, conditions
 				}
 
 			case <-ctx.Done():
-				return lastEvent, wait.ErrWaitTimeout
+				return lastEvent, wait.ErrorInterrupted(nil)
```

Comment:
I think that `ctx.Err()` is better than ` wait.ErrWaitTimeout`.

/cc @deads2k for confirmation.

```suggestion
				return lastEvent, ctx.Err()
```